core/common/src/format/LocalDateFormat.kt

```diff
@@ -21,6 +23,14 @@ public class MonthNames(
 ) {
     init {
         require(names.size == 12) { "Month names must contain exactly 12 elements" }
+        names.indices.forEach { ix ->
```

Comment:
On the second thought about duplicates, this way users would not be able to use single-letter month names for formatting.

Reply:
After an internal discussion, we decided that it's not an issue with the typical use cases for single-letter month names: they are mostly used independently from the other formatting, so you probably won't see things like `2024-J-06` in the wild. Instead, you can find a row of `J`, `F`, `M`, `A`, etc., as labels on a chart or buttons in a calendar app, but then, there is no reason to build a custom format, you can just do something like `date.month.name[0]`.